File: idakonyvtar/src/main/java/eu/eyan/idakonyvtar/controller/IController.java
package eu.eyan.idakonyvtar.controller;

import java.awt.Component;

public interface IController<INPUT, OUTPUT>
{
    Component getView();

    String getTitle();

    void initData(INPUT input);

    void initBindings();

    OUTPUT getOutput();

    Component getComponentForFocus();
}

File: idakonyvtar/src/main/scala/eu/eyan/idakonyvtar/controller/IController.scala
package eu.eyan.idakonyvtar.controller;

import java.awt.Component;

trait IController[INPUT, OUTPUT] {

  def getView(): Component

  def getTitle(): String

  def initData(input: INPUT): Unit

  def initBindings(): Unit

  def getOutput(): OUTPUT

  def getComponentForFocus(): Component
}